content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+```python
+class1 = df.petal_width
+class2 = df.sepal_length
+```
+find variance for each group
+```python
+print(np.var(class1), np.var(class2))
+```
+![var](variancetwosamplettest.png)
+```python
+ratio_check = 0.6811222222222222 / 0.5785315555555559
+
+ratio_check
+```
+![ratiocheck](ratiocheck.png)
+Obviously, the ratio is less than 4:1, thus the variances are assumed to be equal.
+```python
+stats.ttest_ind(a=class1, b=class2, equal_var=True)
+```
+![twosample](twosamplettest.png)
+
+Evaluating this:
+
+**H0:** The mean of the two samples (petal_width and sepal_length) are equal. 
+
+**H1:** The mean of the two samples (petal_width and sepal_length) are not equal.
+
+Our P-value ( p = 3.7974378831185126e-148) is far less than alpha = 0.05, 
+
+Checking of the variance is only making sure that two populations are probably spread out or normally distributed, this will help prevent bias. Hence, the null hypothesis will be assuming that since the two groups are normally distributed, they likely should have the same mean. However, interpreting any form of t-test largely depends on the p-value, the result above explains that the means of the two samples(petal_width and sepal_length) are different. Thus, we will reject the Null Hypothesis which says the mean of the two samples are the same because their variances are assumed to be equal, there is no evidence sustainable enough to prove the two populations have the same mean.
+H1 (alternate hypothesis) is however true.
+
+This allows us to figure out questions like: 
+"Is there a significant or random by the chance difference in the average time spent on mathematics exam from that of English exam?."
+
+#### ANOVA Test
+
+ANOVA is a word coined from `analysis of variance`. It is a statistical concept that shows the differences between the means of more than two independent classes, using variance analysis on samples from those classes.
+
+It is used to check the contrast between three or more samples with one test. Especially when the categorical class has more than two categories.
+
+Why ANOVA? With ANOVA you get to discover obvious differences between the means of your independent features and on getting a clearer picture of the differences, you can get to understand how each of them connects to your dependent variable and what are the influencing factors for the relationship. 
+
+A typical example is, assuming there is an increase in sales for a particular product, and they've tried several advertisement channels so it is difficult to say which one yielded the sales increment the most. They did online posters, billboards, TV ads, mouth-to-mouth, and sponsorship. By carrying out an ANOVA test on the several advertisement channels, they will be able to understand better which one works best out of all channels(independent variables) through the connection it has with the increase in sales which is the dependent variable. It can be very useful in the marketing department of every organization.
```

Comment:
Please use tools like Grammarly to punctuate sentences properly and split all big paragraphs into little ones.

Reply:
done